File: 0073/main.go
package main

func setZeroes(matrix [][]int) {
	if len(matrix) == 0 || len(matrix[0]) == 0 {
		return
	}

	array := []map[int]bool{
		{},
		{},
	}

	i, j, index := 0, 0, 0
	row, column := len(matrix), len(matrix[0])

	for i = 0; i < row; i++ {
		for j = 0; j < column; j++ {
			if matrix[i][j] == 0 {
				array[0][i], array[1][j] = true, true
			}
		}
	}

	for i = range array[0] {
		for index = 0; index < column; index++ {
			matrix[i][index] = 0
		}
	}

	for j = range array[1] {
		for index = 0; index < row; index++ {
			matrix[index][j] = 0
		}
	}
}

// fmt.Println(matrix)

func main() {
	array := [][]int{
		// {1, 1, 1},
		// {1, 0, 1},
		// {1, 1, 1},
		{0, 1, 2, 0},
		{3, 4, 5, 2},
		{1, 3, 1, 5},
	}
	setZeroes(array)
}

File: 0073/main.py
from typing import Dict, List


class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        if len(matrix) == 0 or len(matrix[0]) == 0:
            return

        row, column = len(matrix), len(matrix[0])
        i, j, index = 0, 0, 0

        array: List[Dict[int, bool]] = [{}, {}]

        for i in range(row):
            for j in range(column):
                if matrix[i][j] == 0:
                    array[0][i], array[1][j] = True, True

        for i in array[0]:
            for index in range(column):
                matrix[i][index] = 0
        
        for j in array[1]:
            for index in range(row):
                matrix[index][j] = 0

        
        

        